File: ebean-querybean/src/test/java/org/example/domain/otherpackage/PhoneAttributeConverter.java
package org.example.domain.otherpackage;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter
public class PhoneAttributeConverter implements AttributeConverter<PhoneNumber, String> {
  @Override
  public String convertToDatabaseColumn(final PhoneNumber attribute) {
    return attribute.getMsisdn();
  }

  @Override
  public PhoneNumber convertToEntityAttribute(final String dbData) {
    return new PhoneNumber(dbData);
  }
}

File: tests/test-kotlin/src/main/kotlin/org/example/otherpackage/PhoneAttributeConverter.kt
package org.example.otherpackage

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter
class PhoneAttributeConverter : AttributeConverter<PhoneNumber, String> {
  override fun convertToDatabaseColumn(attribute: PhoneNumber): String {
    return attribute.msisdn
  }

  override fun convertToEntityAttribute(dbData: String): PhoneNumber {
    return PhoneNumber(dbData)
  }
}
